tests/cloud_test_logic/cloud_test_index.py

```diff
@@ -5,113 +5,117 @@ class CloudTestIndex(str, Enum):
     """ Index names that will be mapped to settings of index
     and used in cloud tests,
 
-    Please try to keep names short to avoid hitting name-length limits"""
+    Please try to keep names short to avoid hitting name-length limits
+
+    We create 3 unsructured indexes and 3 structured indexes to test:
+    1) basic_index: a multimodal index, no image preprocessing, no custom model, no text
+    preprocessing. This should be used to test documents with text or image fields.
+
+    2) image_index_with_image_preprocessing: a multimodal index with image processing enabled, this should be used to
+    test document with image fields that requires image processing.
+
+    3) custom_model_index_with_text_preprocessing: a text-only index with custom model and text preprocessing.
+     This should be used to test documents with only text fields that requires text preprocessing.
+
+    For examples,
+        1) You want to test text fields without text preprocessing-> use basic_index
+        2) You want to test image fields without image preprocessing-> use basic_index
+        3) You want to test image fields with image preprocessing -> use image_index_with_image_preprocessing
+        4) You want to test text fields with text preprocessing -> use custom_model_index_with_text_preprocessing
+    """
```

Comment:
helpful docstring.

Reply:
Also, we can get even more variations if we distribute test cases across structured/unstructured. For example:
custom-text-model: unstructured
custom-image-model: structured

Of course this means we don't cover every permutation, but it isn't practical to cover every possible combination during integ tests (and this is the realm of unit tests). We should instead aim to cover a broad set of use cases.

Reply:
ideally, every index also has slightly different infrastructure configurations too